# Ocean Separate -- deciphering Kat's code

In [3]:
# imports
import os
import numpy as np
import xarray

# Load up a file

In [4]:
ex_file = os.path.join(os.getenv('MHW'), 'db', 'extreme_km_dy_by_yr_2019.nc')

In [5]:
ex_ds = xarray.open_dataset(ex_file)
ex_ds

<xarray.Dataset>
Dimensions:  (lat: 720, lon: 1440, year: 38)
Coordinates:
  * lat      (lat) float32 -89.88 -89.62 -89.38 -89.12 ... 89.38 89.62 89.88
  * lon      (lon) float32 0.125 0.375 0.625 0.875 ... 359.1 359.4 359.6 359.9
  * year     (year) int64 1982 1983 1984 1985 1986 ... 2015 2016 2017 2018 2019
Data variables:
    ndays    (lat, lon, year) int32 ...

# NWP

    %% NWP  ij=find(ll(:,:)<66 & ll(:,:)>0 ); ji=find(kk(:,:)>100 & kk(:,:)<180);
    %close all
    mask=NaN*ones(size(voxels,2),size(voxels,3));
    lon=ncread('extreme_dy_by_yr_2019_local_detrend.nc','lon');
    lat=ncread('extreme_dy_by_yr_2019_local_detrend.nc','lat');
    lon=repmat(lon, [1 720]);
    lat=repmat(lat, [1 1440])';

    NWP_vox=voxels(:,401:720,361:624);
    NWP_lon=lon(401:720,361:624);
    NWP_lat=lat(401:720,361:624);
    %figure;pcolor(NWP_lon,NWP_lat,squeeze(NWP_vox(30,:,:)));shading flat;fillmap;
    NWP_vox=permute(NWP_vox,[2 3 1]);
    vox_mean_det(:,2)=squeeze(nanmean(nanmean(NWP_vox)));
    NWP_det=vox_mean_det(:,2);

In [11]:
voxels = ex_ds.ndays.data[:]
voxels.shape

(720, 1440, 38)

In [9]:
lat = ex_ds.lat.data[:]
lon = ex_ds.lon.data[:]
lat.shape, lon.shape

((720,), (1440,))

## Generate grids of lat, lon

In [10]:
lat_grid = np.outer(lat, np.ones(lon.size))
lon_grid = np.outer(np.ones(lat.size), lon)
lat_grid.shape

(720, 1440)

## Cut on Region

In [14]:
NWP_vox=voxels[401:720,361:624, :]
NWP_lon=lon_grid[401:720,361:624]
NWP_lat=lat_grid[401:720,361:624]

In [15]:
NWP_vox.shape

(319, 263, 38)

## Mean

In [16]:
NWP_vox=permute(NWP_vox,[2 3 1]);
vox_mean_det(:,2)=squeeze(nanmean(nanmean(NWP_vox)))

SyntaxError: invalid syntax (671427385.py, line 1)

In [20]:
NWP_mean = np.nanmean(NWP_vox, axis=(0,1))
NWP_mean

array([0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0.])

In [22]:
np.max(NWP_vox[...,0])

0

----

In [24]:
voxels[201,361,:]

array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], dtype=int32)

In [25]:
np.max(voxels)

0